components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -852,6 +833,102 @@ public ServiceProvider getServiceProviderByClientId(String clientId, String clie
+
+        ImporterResponse importerResponse = new ImporterResponse();
+        ServiceProvider serviceProvider = unmarshalSP(content, fileName, tenantDomain);
+        if (serviceProvider == null) {
+            throw new IdentityApplicationManagementException(String.format("Empty Service Provider file %s upload by" +
+                    " tenant: %s", fileName, tenantDomain));
+        }
+
+
+        Collection<ApplicationMgtListener> listeners =
+                ApplicationMgtListenerServiceComponent.getApplicationMgtListeners();
+
+        ServiceProvider savedSP = null;
+        try {
+            if (isUpdate) {
+                savedSP = getApplicationExcludingFileBasedSPs(serviceProvider.getApplicationName(), tenantDomain);
+                if (savedSP == null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("Service provider %s@%s is not found, so creating new Service provider",
+                                serviceProvider.getApplicationName(), tenantDomain));
+                    }
+                    isUpdate = false;
+                }
+            }
+
+            if (!isUpdate) {
+                ServiceProvider basicApplication = new ServiceProvider();
+                basicApplication.setApplicationName(serviceProvider.getApplicationName());
+                basicApplication.setDescription(serviceProvider.getDescription());
+                savedSP = addApplication(basicApplication, tenantDomain, username);
+            }
+            serviceProvider.setApplicationID(savedSP.getApplicationID());
+            serviceProvider.setOwner(getUser(tenantDomain, username));
+
+            for (ApplicationMgtListener listener : listeners) {
+                if (listener.isEnable()) {
+                    listener.doPreUpdateApplication(serviceProvider, tenantDomain, username);
+                }
+            }
+
+            for (ApplicationMgtListener listener : listeners) {
+                if (listener.isEnable()) {
+                    listener.doImportServiceProvider(serviceProvider);
+                }
+            }
+
+            isImportSP.set(true);
+            try {
+                updateApplication(serviceProvider, tenantDomain, username);
+            } catch (IdentityApplicationManagementValidationException e) {
+                importerResponse.setApplicationName(null);
+                importerResponse.setErrors(e.getValidationMsg());
+                if (!isUpdate) {
+                    deleteSavedSP(savedSP, tenantDomain, username);
+                }
+                return importerResponse;
+            } finally {
+                isImportSP.set(false);
+            }
+            importerResponse.setApplicationName(serviceProvider.getApplicationName());
+            importerResponse.setErrors(new String[0]);
+            return importerResponse;
+        } catch (IdentityApplicationManagementValidationException e) {
```

Comment:
Is there a reason not to log the stack trace ?

Reply:
Yes, these are validation errors during the import. These errors will be displayed in the UI.